dts/bindings/mtd/atmel,sam-flash.yaml

```diff
@@ -0,0 +1,18 @@
+description: Atmel SAM flash area
+
+compatible: "atmel,sam-flash"
```

Comment:
provide full documentation with examples

Reply:
Agreed, wanted to get some feedback first :)

Reply:
See https://github.com/zephyrproject-rtos/zephyr/compare/550e9ddd81fe23b5d73f7bed237df1c3803ae1f2..1c485944b0a69e4e06bc205ab0eee554ab3a2b03